flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java

```diff
@@ -225,7 +226,9 @@ protected void run(String[] args) throws Exception {
 			final ClusterClient<T> client;
 
 			// directly deploy the job if the cluster is started in job mode and detached
-			if (flip6 && clusterId == null && runOptions.getDetachedMode()) {
+			boolean detachedMode = customCommandLine.isDetachedMode(commandLine);
+			LOG.debug("Detached job mode is set to {}", detachedMode);
+			if (flip6 && clusterId == null && detachedMode) {
```

Comment:
The `RunOptions` should already contain the information whether we are in detached mode or not.

Reply:
not in the case of yarn. `RunOptions` only evaluates `-d`, not `-yd`.

Reply:
Yes, I think there should not be any `-yd`.

Reply:
I agree that we shouldn't have `-yd`, but it is part of the client API and unless we want to change existing behavior we can't rely on `RunOptions` alone.

Reply:
I would be willing to do so. Alternatively, we could register a `-yd` option in `CliFrontend` for backwards compatibility reasons. This would, however, also have an effect if used with standalone mode.